### PR TITLE
Avoid two allocations

### PR DESCRIPTION
**Pull request by Rawk**

### CODE REVIEW

File: src/comment.rs [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 use std::{borrow::Cow, iter};
 
-use itertools::{MultiPeek, multipeek};
+use itertools::{Itertools as _, MultiPeek, multipeek};
```
Comment (ytmimi) [inline src/comment.rs:5]: Could you explain why you're brining `Itertools` into scope.

Comment (Rawk) [reply src/comment.rs:5]: It is used for the `Itertools::join` on line 1077.

File: src/comment.rs [hunk 2]
```diff
@@ -1075,8 +1074,7 @@ fn light_rewrite_comment(
             // Preserve markdown's double-space line break syntax in doc comment.
             trim_end_unless_two_whitespaces(left_trimmed, is_doc_comment)
         })
-        .collect();
-    lines.join(&format!("\n{}", offset.to_string(config)))
+        .join(&format!("\n{}", offset.to_string(config)))
```
Comment (ytmimi) [inline src/comment.rs:1077]: Okay, so just want to make sure I'm understanding this correctly. This avoids allocating the `Vec<&str>` and directly allocates into a `String`?

Comment (Rawk) [reply src/comment.rs:1077]: Yes, exactly.

**Review by ytmimi — APPROVED** [review]

**Pull request merged**